test/dynamo/test_comptime.py

```diff
@@ -0,0 +1,216 @@
+# Owner(s): ["module: dynamo"]
+
+import re
+from io import StringIO
+
+import torch._dynamo.test_case
+import torch._dynamo.testing
+from torch._dynamo.eval_frame import comptime
+
+# Because we don't support free variables in comptime at the moment,
+# we have to communicate via globals.  This also means these tests cannot
+# be run in parallel in a single process (not that you'd... ever want
+# to do that?)
+FILE = None
+SELF = None
+
+
+class ComptimeTests(torch._dynamo.test_case.TestCase):
+    def test_print_graph(self):
+        global FILE
+        FILE = StringIO()
+
+        @torch._dynamo.optimize("eager")
+        def f(x):
+            y = x * 2
+
+            @comptime
+            def _(ctx):
+                ctx.print_graph(verbose=False, file=FILE)
+
+            return y + 3
+
+        f(torch.randn(2))
+        self.assertExpectedInline(
+            FILE.getvalue().strip(),
+            """\
+def forward(self, x : torch.Tensor):
+    mul = x * 2;  x = None""",
+        )
+
+    def test_print_disas(self):
+        global FILE
+        FILE = StringIO()
+
+        @torch._dynamo.optimize("eager")
+        def f(x):
+            y = x * 2
+
+            @comptime
+            def _(ctx):
+                ctx.print_disas(file=FILE)
```

Comment:
Perhaps we should expose some of these through a more compact API like `dynamo.debug.print_disas()` (which would just be this same function) -- I'd expect a lot of common cases will just be calling a single one of these functions.

Reply:
The one subtlety is that we would have to make all of these functions special forms in Dynamo, in the same way `comptime` has special handling. We can still do that though. Another possibility is to just define the function as

```
def print_disas(ctx):
    ctx.print_disas()
```

and then you can simply say `comptime(torch._dynamo.debug.print_disas)` to do the single line, without having to introduce other special forms (but you'd still have to import modules for two identifiers.) These are both easy to implement so it's mostly a matter of what people prefer.

Reply:
Actually, I guess if you don't mind introducing a stack frame (which does matter in a few cases), you can just use functional abstraction. I'll go add these.